### PR TITLE
New endpoint for displaying user scores

### PR DESCRIPTION
**Pull request by IuliaVele**
Fixes: [https://github.com/AACraiu/golfr_frontend/issues/1]

**Changes**

New endpoint for displaying user scores.

**Before**

-

**After**
endpoint: `/api/golfers/{userId}` (in this case 1)

Screenshot 2023-08-23 at 17 34 38

### CODE REVIEW

File: app/controllers/api/scores_controller.rb [hunk 1]
```diff
@@ -37,6 +37,16 @@ def destroy
       }
     end
 
+    def user_scores
+      score = Score.where(user_id: params[:user_id])
```
Comment (sergiuapostugg) [inline app/controllers/api/scores_controller.rb]: Can we order them descending by their play date? Just like we do for the feed.

Don't forget about n+1 queries!

Comment (IuliaVele) [reply app/controllers/api/scores_controller.rb]: Sure!

**Comment by sergiuapostugg** [comment]
Please write automated tests for the new endpoint.

File: spec/controllers/api/scores_controller_spec.rb [hunk 2]
```diff
@@ -104,4 +104,26 @@
...
   end
+
+  describe 'GET user_scores' do
+    it 'should return the scores related to the user' do
+      @user = create(:user, name: 'User', email: 'user7@email.com', password: 'userpass')
+
+      sign_in(@user, scope: :user)
+
+      @score1 = create(:score, user: @user, total_score: 79, played_at: '2021-05-20')
+      @score2 = create(:score, user: @user, total_score: 59, played_at: '2021-05-25')
```
Comment (sergiuapostugg) [inline spec/controllers/api/scores_controller_spec.rb]: Why did you choose to define another user? You already have user1 and user2.

File: spec/controllers/api/scores_controller_spec.rb [hunk 3]
```diff
@@ -104,4 +104,26 @@
...
+
+      get api_golfers_path(user_id: @user.id)
```
Comment (sergiuapostugg) [inline spec/controllers/api/scores_controller_spec.rb]: Please write another test for the case when you request an user's scores, logged in as another user.

File: spec/controllers/api/scores_controller_spec.rb (hunk 2 above)
Comment (IuliaVele) [reply spec/controllers/api/scores_controller_spec.rb]: I've changed this to use the user1 defined in before each block.

**Review by sergiuapostugg — APPROVED** [review]

**Pull request merged**